janitor/functions/count_cumulative_unique.py

```diff
@@ -12,73 +15,85 @@ def count_cumulative_unique(
-        import janitor
-
-        df = pd.DataFrame(...)
-
-        df = df.count_cumulative_unique(
-            column_name='animals',
-            dest_column_name='animals_unique_count',
-            case_sensitive=True
-        )
-    ```
-
     A new column will be created containing a running
     count of unique values in the specified column.
     If `case_sensitive` is `True`, then the case of
     any letters will matter (i.e., `a != A`);
     otherwise, the case of any letters will not matter.
 
-    This method mutates the original DataFrame.
-
-    :param df: A pandas dataframe.
-    :param column_name: Name of the column containing
-        values from which a running count of unique values
-        will be created.
+    This method does not mutate the original DataFrame.
+
+    Examples:
+
+        >>> import pandas as pd
+        >>> import janitor
+        >>> df = pd.DataFrame({
+        ...     "letters": list("aabABb"),
+        ...     "numbers": range(4, 10),
+        ... })
+        >>> df
+          letters  numbers
+        0       a        4
+        1       a        5
+        2       b        6
+        3       A        7
+        4       B        8
+        5       b        9
+        >>> df.count_cumulative_unique(
+        ...     column_name="letters",
+        ...     dest_column_name="letters_unique_count",
+        ... )
+          letters  numbers  letters_unique_count
+        0       a        4                     1
+        1       a        5                     1
+        2       b        6                     2
+        3       A        7                     3
+        4       B        8                     4
+        5       b        9                     4
+
+    Example: Cumulative counts, ignoring casing.
+
+        >>> df.count_cumulative_unique(
+        ...     column_name="letters",
+        ...     dest_column_name="letters_unique_count",
+        ...     case_sensitive=False,
+        ... )
+          letters  numbers  letters_unique_count
+        0       a        4                     1
+        1       a        5                     1
+        2       b        6                     2
+        3       A        7                     2
+        4       B        8                     2
+        5       b        9                     2
+
+    :param df: A pandas DataFrame.
+    :param column_name: Name of the column containing values from which a
+        running count of unique values will be created.
     :param dest_column_name: The name of the new column containing the
         cumulative count of unique values that will be created.
     :param case_sensitive: Whether or not uppercase and lowercase letters
-        will be considered equal (e.g., 'A' != 'a' if `True`).
-
+        will be considered equal.
     :returns: A pandas DataFrame with a new column containing a cumulative
         count of unique values from another column.
     """
+    check_column(df, column_name)
 
```

Comment:
can we add a check here for the dest_column_name, something like `check_column(df, dest_column_name, present=True)` to raise an error if `dest_column_name` already exists.